doc/content/xapi/migration_overview.md

```diff
@@ -0,0 +1,83 @@
+---
+title: From client RPC migration request to xapi internals
+---
+
+# How the migration request goes through Xen API
+
+## Overview
+
+In this document we will utilize the `VM.pool_migrate` request to illustrate
+the interaction between various components within the XAPI toolstack during
+migration. However this schema can be applied to other requests as well.
+
+Not all parts of the Xapi toolstack are shown here as not all are involved in
+the migration process. For instance you won't see the ***squeezed***
+nor ***mpathalert*** two daemons that belong to the toolstack but don't
+participate in the migration of a VM.
+
+## Anatomy of a VM migration
+
+- Migration is initiated by a Xapi client that sends `VM.pool_migrate`, an RPC XML request.
+- The Xen API server handles this request and dispatches it to the server.
+- The server is generated using `XAPI IDL` and requests are wrapped whitin a
+context, either to be forwared to a host or executed locally. Broadly, the
+context follows RBAC rules. The executed function is related to the message of
+the request (refer to [XenAPI Reference](https://xapi-project.github.io/xen-api/)).
+- In the case of the migration you can refer to *ocaml/idl/datamodel_vm.ml*.
+- The server will dispatch the operation to server helpers, executing the
+operation synchronously or asynchronously and returning the RPC answer.
+- Operations can then be forwarded or executed locally.
+- When executed locally the migration operation is split into several
+operations which can run by the *xenopsd backend*. The steps involve:
```

Comment:
There is room for confusion here: might want to say "the xenopsd backend in xapi` in contrast to the xenopsd daemon which receives its requests from xapi via message switch. Xenopsd translates a high-level request from xapi into (abstract) atomic operations and executes them.

Reply:
Yes and this is the part where I'm little bit lost ;).
 So IIUC the high-level request are send by Xapi (xapi_xenops.ml) and posted into a qeue of the message switch. Then the "xenopsd daemon" read this message and split it into several "atomic" operations before executing them by calling the "xenops backend in xapi". 
So the exchange of high level operation between xapi and xenopsd daemon is done via the message switch right?
I thought that the message switch was used to post atomic operations after xenopsd split the high-level ones into atomics. And I thought that high-level operations are send to xenopsd daemon via XML RPC.
I will fix that part for sure and also the flowchart that is not correct.
The flowchart can be pre-viewed using [mermaid live](https://mermaid.live/edit#pako:eNqFVW1r2zAQ_iuHR0dimrSM9YsZgzRNoNBAtpYSVpdUsZVYVJY8SW4a2v73nSzLTpzC9CWW9Nxzz73o8hYkMqVBFMRizeU2yYgycHcZC8B1cgJTprSBLQWiKGiDt0xsYLUDAovR_BoSzqgwYDJigGnQVKQWQAQsZjeD3_MxKPq3pNo4Rgfv9RakYPWm34fB4N0aarB4tPM274PBT2dn1yvaLDVVL1S9xcHTFdMFMUlmjVrQ_grD1mKY8zB8ioMPB9Vmx6kXr42SzzT6Mh5NJ5PJqdsOtiw1WfS9eG1yUQV8JyXHPCTPwIShShCua8pytVGkyCAOqugaYBy08pxfJ6nxezmajs7Pj_z6g5RorIoiuwgu4OLTfNgMpj4fCeEceozC_WxYoIhlzlCYoX2bztp3DxM4Ko2EDRXUXqZQalu48PrqJjxOZxgeprG_F1KrQC4bEf_xv8woL6jSD6ijJqhPWmLv02O978eubw-wGlz0ORFkQ3NbXCukytRL7oUgkdNijxqklXI_g9kmN60Dzp4pCjkm6Co5oKqEjK2QTzr4lQpZVIHfKSL0Wqoc3Nl-XBR6tVu8K6QLvm99VuRzia9yLTk-WVs1LBrk-I6x7tbU0S2rBl2TxOqtlNQXad01e_7a3q0h-01bST-whB8oYl7q7GwuC5ACpNoMETK0ioc1dphwojVLDnPQJVzZHlcPZzOqNaYP9JZh_0Twq6Q4AuL48XMdK3xYODKcEFmcWTEuxg5za45wt2k-7Jt2YPB8OMNIkqAWm1ZkwkmW2l9tpKLOqiMBK-lJLt1JHIvFGHqcrRCaGMX7ddkcMxjpi7HMdti4L0xL5WigPYCvrV_faW58HBq2Y2QymXbHyLe98TXGoWzHrJ-uqAMfijY-R0ctjK25JSqtxDrgQz3b3A4y7MKOtvqm0XQ-OpprwWmQU5UTluLfzps1jgOTodc4iPAzpWtScmMb8AOhBKfU7U4kQWRUSU-DskjxBV4xgg2bB9Ea5y-eFkT8kdLvP_4B5JJBnw)

Reply:
I updated the doc

Reply:
The splitting of high-level operations into atomic operations happens inside xenopsd and there is no communication with xapi about this. I tried to clarify the language about "xenopsd backend". There is a part in xapi that communicates with xenopsd (which could be thought of as backend) and there is xenopsd itself (which used to have different backends in itself but no longer does).

Reply:
Ok. Thanks a lot for the help :+1: .  I will update the flowgraph. I think I have the right picture now.